### DISCUSSION TITLE
Keyboard Quantizer B を接続するとキーボードが動かない

### DISCUSSION THREAD

**Discussion by ys0715** [Yowkees/keyball#155] [Q&A]
Keyball61を先日購入し便利に使用させていただいています。
無線化して使用してみたいと思いKeyboard Quantizer Bを接続してみたところ、キー入力・トラックボール両方ともうまく動きませんでした。
Keyball61をPCに直接接続した時はUSB接続した側のOLEDにはキー入力やトラックボールの情報が表示されもう片側のOLEDにはKeyballロゴが表示されますが、Keyboard Quantizer Bを接続した場合、両方のOLEDにKeyballロゴが表示されています。
恐らくではありますが、左右自動判定がうまく動かずにその状況になっているのかなと想像しているのですが、Keyball61のファームウェアをみても該当部分がよく分からりませんでした。
Keyboard Quantizer Bのファームウェアを改造した方が早い気もしますが、そちらも正直あまり理解していないので…
もし何か解決策や情報などがございましたら共有して頂けるととても助かります。
よろしくお願いいたします。

**Comment by Yowkees** [comment]
大変申し訳ありません。私がKeyboard Quantizer Bを使用したことがないため、どのような原因でキー入力が機能しないのかアドバイスが難しいです。
左右自動判定がうまくいかなくても何らかのキー入力はできるはずですので、接続自体がうまくいっていない印象ですが、確かではありません。申し訳ありませんがよろしくお願いします。

**Comment by ys0715** [comment]
返信ありがとうございます。

同じケーブル、同じKeyboard Quantizer Bで他のキーボード(cocot46・uzu42)ではちゃんと動いているので、ファームウェアに原因が有るのかなと考え、ファームウェアの違いというと左右自動判定とトラックボール関係くらいなのかなと思い質問させていただきました。

ひとまず現状では動かせないということでBLE化は諦め、有線で使用していくことにします。

もし今後、何らかのヒントやこうやったら動いたよ！等の情報が有りましたら、その際は共有して頂けると嬉しいです。

それではよろしくお願いいたします。

**Comment by UGKEN** [comment]
横からすみません。
こちらの件について、自分もKeyboard Quantizer Bを所持しており、同じくKeyball39だけが認識しないという問題が発生しています。

Webで「Keyball39 認識しない」と言ったワードで検索してみたところ、同じ原因かわかりませんが特定環境のMacで同様に認識しないことがあるようで、そのケースではSPLIT_USB_TIMEOUTの値を増やすことで対処できたそうです。
本件についても同じ対処で対応可能かもしれません。
Macの場合、SPLIT_USB_TIMEOUTを2500程度に増やすと認識するようになるそうです。

ただ、自分のほうは別件で報告したビルドエラーが発生するためファームウェアがビルドできず、実際にこの対処で対応可能なのかは検証できておりません。

もしKeyboard Quantizer BとKeyballシリーズを所持しており、ファームウェアのビルドが可能な方がいらっしゃいましたら、SPLIT_USB_TIMEOUTの値を2500程度に上げてテストしていただけるとありがたいです。

**Comment by UGKEN** [comment]
本件ですが、先程ファームウェアがビルドできない問題が解決しましたのでSPLIT_USB_TIMEOUTを2500に増やしてみたところ、無事にKeyboard Quantizer BにてKeyball39が認識するようになりました。

Keyboard Quantizer Bに認識させるためにはSPLIT_USB_TIMEOUTの値が小さいことが問題であったようです。
